### ISSUE TITLE
Async simulation core ignores lifetimes, which leads to "use after free".

### ISSUE THREAD

**Issue by egnees** [osukhoroslov/dslab#338]
```rust

use std::{cell::RefCell, rc::Rc};

use dslab_core::{Event, EventHandler, Simulation};
use serde::Serialize;

#[derive(Default)]
struct Component {}

impl EventHandler for Component {
    fn on(&mut self, _event: Event) {}
}

#[derive(Serialize, Clone)]
struct Info {
    info: String,
}

impl Drop for Info {
    fn drop(&mut self) {
        println!("dropped info: {:?}", self.info);
    }
}

fn main() {
    let mut sys = Simulation::new(12345);

    let comp = Rc::new(RefCell::new(Component::default()));
    sys.add_handler("comp", comp);
    let ctx = sys.create_context("comp");

    {
        let info = Info { info: "1".to_string() };
        let info_ref = &info;

        ctx.spawn(async move {
            println!("future started...");
            let info = info_ref.clone();
            println!("info: {:?}", info.info);
        });
    }

    sys.step_until_no_events();
}


```

The code above compiles, but should not in the safe rust.
The output is the following:
```
dropped info: "1"
future started...
info: "\0"
dropped info: "\0"
```

**Comment by nogokama** [comment]
👀

**Comment by osukhoroslov** [comment]
Resolved in #340
